qiskit/qpy/binary_io/circuits.py

```diff
@@ -149,7 +149,7 @@ def _read_instruction_parameter(file_obj, version, vectors):
     return param
 
 
-def _read_instruction(file_obj, circuit, registers, custom_instructions, version, vectors):
+def _read_instruction(file_obj, circuit, registers, custom_operations, version, vectors):
```

Comment:
Was it necessary to do all this renaming? This is going to cause merge conflict headaches for #8055 and #7300

Reply:
We decided to keep this offline, since the merge-conflict damage should already have been done (I just had to fix it in this PR), and the new `CircuitInstruction` name makes the names confusing if the names are left in the pre-PR state.